### PR TITLE
Moved/Improved g.nodes.to_df and g.edges.to_df and added support for python312

### LINKED ISSUE #1549: extend to_df fns to remove all jsons and have column store

### PR DESCRIPTION
**Pull request by Haaroon**
### What changes were proposed in this pull request?

 Moved/Improved g.nodes.to_df and g.edges.to_df and added support for python312 #1537 

### Issues

resolves #1549 #1548

### CODE REVIEW

File: raphtory/src/python/graph/views/graph_view_modules/export.rs [hunk 1]
```diff
@@ -1,16 +1,236 @@
...
+                } else {
+                    name.to_string()
+                };
+                let _ =
+                    properties_map.insert(column_name, t_prop.latest().unwrap_or(Prop::from("")));
+            });
+        }
+    }
+
+    fn convert_timestamp(time: i64) -> NaiveDateTime {
```
Comment (miratepuffin) [inline raphtory/src/python/graph/views/graph_view_modules/export.rs:128]: Can this be changed to include a timezone - history/earliest_time etc do this

File: raphtory/src/python/graph/views/graph_view_modules/export.rs [hunk 2]
```diff
@@ -1,16 +1,236 @@
...
+            });
+        column_names.push("update_histories".parse().unwrap());
+        is_prop_both_temp_and_const
+    }
+
+    fn create_row(
+        convert_datetime: bool,
+        explode: bool,
+        column_names: &Vec<String>,
+        mut properties_map: HashMap<String, Prop>,
```
Comment (miratepuffin) [inline raphtory/src/python/graph/views/graph_view_modules/export.rs:174]: Doesn't need to be mut

File: raphtory/src/python/graph/views/graph_view_modules/export.rs [hunk 3]
```diff
@@ -19,57 +239,74 @@ impl PyGraphView {
...
-    /// - "property_history": The history of the node's properties. This column will be included if both `include_node_properties` and `include_property_histories` are set to `true`.
-    /// - "update_history": The update history of the node. This column will be included if `include_update_history` is set to `true`.
+    /// - "update_history": The update history of the node.
     ///
     /// Args:
-    ///     include_node_properties (bool): A boolean wrapped in an Option. If set to `true`, the "properties" and "property_history" columns will be included in the DataFrame. Defaults to `true`.
-    ///     include_update_history (bool): A boolean wrapped in an Option. If set to `true`, the "update_history" column will be included in the DataFrame. Defaults to `true`.
-    ///     include_property_histories (bool): A boolean wrapped in an Option. If set to `true`, the "property_history" column will be included in the DataFrame. Defaults to `true`.
+    ///     include_property_histories (bool): A boolean, if set to `true`, the history of each property is included, if `false`, only the latest value is shown.  Defaults to `true`.
```
Comment (miratepuffin) [inline raphtory/src/python/graph/views/graph_view_modules/export.rs:245]: Should note that this gets ignored when exploding

File: raphtory/src/python/graph/views/graph_view_modules/export.rs [hunk 4]
```diff
@@ -85,77 +322,79 @@ impl PyGraphView {
     /// - "update_history_exploded": The exploded update history of the edge. This column will be included if `explode_edges` is set to `true`.
     ///
     /// Args:
-    ///     explode_edges (bool): A boolean wrapped in an Option. If set to `true`, the "update_history_exploded" column will be included in the DataFrame. Defaults to `false`.
-    ///     include_edge_properties (bool): A boolean wrapped in an Option. If set to `true`, the "properties" and "property_histories" columns will be included in the DataFrame. Defaults to `true`.
-    ///     include_update_history (bool): A boolean wrapped in an Option. If set to `true`, the "update_history" column will be included in the DataFrame. Defaults to `true`.
-    ///     include_property_histories (bool): A boolean wrapped in an Option. If set to `true`, the "property_histories" column will be included in the DataFrame. Defaults to `true`.
+    ///     include_property_histories (bool): A boolean, if set to `true`, the history of each property is included, if `false`, only the latest value is shown.  Defaults to `true`.
```
Comment (miratepuffin) [inline raphtory/src/python/graph/views/graph_view_modules/export.rs:325]: Should note that this is ignored when exploding

**Review by miratepuffin — CHANGES_REQUESTED** [review]
Can you also add these functions onto the edges and nodes object as `.to_df()`

I think we need to make these go significantly faster - but lets get the new version out first.

**Comment by Haaroon** [comment]
> Can you also add these functions onto the edges and nodes object as `.to_df()`
> 
> I think we need to make these go significantly faster - but lets get the new version out first.

yes done but instead i just moved it to to_df entirely

File: raphtory/src/python/graph/edges.rs [hunk 5]
```diff
@@ -228,6 +239,94 @@ impl PyEdges {
...
         (move || edges.layer_names().map(|e| e.collect_vec())).into()
     }
+
+    /// Converts the graph's edges into a Pandas DataFrame.
+    ///
+    /// This method will create a DataFrame with the following columns:
+    /// - "src": The source node of the edge.
+    /// - "dst": The destination node of the edge.
+    /// - "layer": The layer of the edge.
+    /// - "properties": The properties of the edge. This column will be included if `include_edge_properties` is set to `true`.
```
Comment (miratepuffin) [inline raphtory/src/python/graph/edges.rs:249]: Think this comment is slightly out of date

File: raphtory/src/python/graph/node.rs [hunk 6]
```diff
@@ -566,6 +572,80 @@ impl PyNodes {
             .get(node)
             .ok_or_else(|| PyIndexError::new_err("Node does not exist"))
     }
+
+    /// Converts the graph's nodes into a Pandas DataFrame.
+    ///
+    /// This method will create a DataFrame with the following columns:
+    /// - "name": The name of the node.
+    /// - "properties": The properties of the node. This column will be included if `include_node_properties` is set to `true`.
```
Comment (miratepuffin) [inline raphtory/src/python/graph/node.rs:580]: This comment is also slightly out of date

**Review by miratepuffin — APPROVED** [review]

**Pull request merged**